From a5206e576cea73f0d63cdc76f3be712613cacfc0 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Fri, 31 Jul 2015 11:29:11 +0700
Subject: [PATCH 1/4] qtmultimedia: add configure flag for playbin not using
 videoconvert Add GST_PLAY_FLAG_NATIVE_VIDEO into
 playbin flags for not using videoconvert Qtrenderersink
 can display NV12 format from omxh264dec, so
 videoconvert is not needed

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 .../mediaplayer/qgstreamerplayersession.cpp        |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp b/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
index 4dce341..032498b 100644
--- a/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
+++ b/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
@@ -153,7 +153,7 @@ QGstreamerPlayerSession::QGstreamerPlayerSession(QObject *parent)
         int flags = GST_PLAY_FLAG_VIDEO | GST_PLAY_FLAG_AUDIO |
                     GST_PLAY_FLAG_NATIVE_VIDEO | GST_PLAY_FLAG_NATIVE_AUDIO;
 #else
-        int flags = GST_PLAY_FLAG_VIDEO | GST_PLAY_FLAG_AUDIO;
+        int flags = GST_PLAY_FLAG_VIDEO | GST_PLAY_FLAG_AUDIO | GST_PLAY_FLAG_NATIVE_VIDEO;
         QByteArray envFlags = qgetenv("QT_GSTREAMER_PLAYBIN_FLAGS");
         if (!envFlags.isEmpty()) {
             flags |= envFlags.toInt();
-- 
1.7.9.5

